src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueTests.cs

```diff
@@ -3511,77 +3539,87 @@ class C
+                    Row(baseParameterCount + generation * 2, TableIndex.Param, EditAndContinueOperation.Default),
+                    hasAttribute ? Row(baseAttributeCount + generation, TableIndex.CustomAttribute, EditAndContinueOperation.Default) : default);  // adding a new attribute row for attribute on C#* definition
+
+                CheckEncMapDefinitions(reader,
+                    Handle(baseTypeCount + generation, TableIndex.TypeDef),
+                    Handle(baseMethodCount + generation * 2 - 1, TableIndex.MethodDef),
+                    Handle(baseMethodCount + generation * 2, TableIndex.MethodDef),
+                    Handle(baseParameterCount + generation * 2 - 1, TableIndex.Param),
+                    Handle(baseParameterCount + generation * 2, TableIndex.Param),
+                    hasAttribute ? Handle(baseAttributeCount + generation, TableIndex.CustomAttribute) : default);
+
+                var newTypeDefHandle = reader.TypeDefinitions.Single();
+                var newTypeDef = reader.GetTypeDefinition(newTypeDefHandle);
+                CheckStringValue(readers, newTypeDef.Name, "C#" + generation);
+
+                if (hasAttribute)
+                {
+                    var attribute = reader.GetCustomAttribute(reader.CustomAttributes.Single());
 
-            CheckEncLogDefinitions(reader2,
-                Row(4, TableIndex.TypeDef, EditAndContinueOperation.Default),
-                Row(4, TableIndex.TypeDef, EditAndContinueOperation.AddMethod),
-                Row(5, TableIndex.MethodDef, EditAndContinueOperation.Default),
-                Row(4, TableIndex.TypeDef, EditAndContinueOperation.AddMethod),
-                Row(6, TableIndex.MethodDef, EditAndContinueOperation.Default),
-                Row(5, TableIndex.MethodDef, EditAndContinueOperation.AddParameter),
-                Row(4, TableIndex.Param, EditAndContinueOperation.Default),
-                Row(5, TableIndex.MethodDef, EditAndContinueOperation.AddParameter),
-                Row(5, TableIndex.Param, EditAndContinueOperation.Default));
+                    // parent should be C#1
+                    var aggregator = GetAggregator(readers);
+                    var parentHandle = aggregator.GetGenerationHandle(attribute.Parent, out var parentGeneration);
+                    Assert.Equal(generation, parentGeneration);
+                    Assert.Equal(newTypeDefHandle, parentHandle);
 
-            CheckEncMapDefinitions(reader2,
-                Handle(4, TableIndex.TypeDef),
-                Handle(5, TableIndex.MethodDef),
-                Handle(6, TableIndex.MethodDef),
-                Handle(4, TableIndex.Param),
-                Handle(5, TableIndex.Param));
+                    // the serialized type name should be the base name, not `C#1`
+                    CheckBlobValue(readers, attribute.Value, new byte[] { 0x01, 0x00, 0x01, (byte)'C', 0x00, 0x00 });
```

Comment:
>new byte[] { 0x01, 0x00, 0x01, (byte)'C', 0x00, 0x00 } [](http://example.com/codeflow?start=61&length=54)

My decoding of binary blobs for attributes is rusty ;-) It's hard to tell whether this is correct. Consider adding a test decoding method

Reply:
Decoding general attribute is not trivial amount of code. 
Here the important part is that only the name `C` is encoded in the attribute rather then `C#1`